File: geotrellis-python/geotrellis/spark/io/file/LayerPath.py
import os.path

class _LayerPath(object):
    def __call__(self,  first, second = None):
        if second is None:
            layerid = first
            return _namezoom(layerid)
        else:
            catalogPath = first
            layerid = second
            return os.path.join(catalogPath, _namezoom(layerid))

def _namezoom(layerid):
    return "{name}/{zoom}".format(name = layerid.name, zoom = layerid.zoom)

LayerPath = _LayerPath()

File: layers/src/main/scala/geotrellis/layers/file/LayerPath.scala
package geotrellis.layers.file

import java.io.File

import geotrellis.layers.LayerId

object LayerPath {
  def apply(catalogPath: String, layerId: LayerId): String =
    new File(catalogPath, apply(layerId)).getAbsolutePath
  def apply(layerId: LayerId): String = s"${layerId.name}/${layerId.zoom}"
}
